tests/LinAlg/vectorTests.hpp

```diff
@@ -209,15 +209,15 @@ class VectorTests : public TestBase
     assert(Nx > Nfrom);
     x.setToConstant(two);
 
-    real_type* from_buffer = createLocalBuffer(Nx, one);
```

Comment:
In addressing issue #133, I was trying to remove the function `createLocalBuffer`. If I were to use the existing implementation of `hiop::LinearAlgebraFactory::createRawArray`, there is no way to initialise device memory to a default value in an implementation agnostic way.

The solutions to this that I could come up with were:
- Leave methods `createLocalBuffer` and `deleteLocalBuffer` as is in implementation-specific tests
- Add a default value to `hiop::LinearAlgebraFactory::createRawArray` for memory to be initialised to

I supported the latter, as it seemed like a relatively easy change to implement, that would not affect any other areas of code.

Instead of how I have implemented it currently, we could write an overload of the `hiop::LinearAlgebraFactory::createRawArray` method that takes two arguments, so as to avoid using `std::fill` when it is unnecessary.

Reply:
Could you just add `fillArray(T*,T);` helper method to test framework?

Reply:
The existing `createLocalBuffer` method in unit testing already has the functionality of what `fillArray` would accomplish. 

It's seeming more as though this is just an unnecessary change that doesn't add any useful functionality to src, only being useful in testing. Because of this, it's probably fine just to leave this as is.